src/GraphQL/Types/Composite/ComplexGraphType.cs

```diff
@@ -210,16 +210,27 @@ public virtual FieldType AddField(FieldType fieldType)
             string? deprecationReason = null)
             where TGraphType : IGraphType
         {
+            IFieldResolver? resolver = null;
+
+            if (resolve != null)
+            {
+                // create an instance expression that points to the instance represented by the delegate
+                // for instance, if the delegate represents obj.MyMethod,
+                // then the lambda would be: _ => obj
+                var param = Expression.Parameter(typeof(IResolveFieldContext), "context");
+                var body = Expression.Constant(resolve.Method.IsStatic ? null : resolve.Target, resolve.Method.DeclaringType);
```

Comment:
```suggestion
                var body = Expression.Constant(resolve.Target, resolve.Method.DeclaringType);
```
![изображение](https://user-images.githubusercontent.com/21261007/156267768-cb009e3e-930b-4242-a1ee-6980170a3b02.png)